api/db-migrations/000012_add_traffic_rule_name.up.sql

```diff
@@ -0,0 +1,42 @@
+UPDATE router_versions
+SET traffic_rules = combined_rule FROM (
+    SELECT
+        id
+        , json_agg(rule_with_name) AS combined_rule
+    FROM
+        (
+            SELECT
+                t1.id
+                , t1.position
+                , t1.name
+                , t2.elem
+                , jsonb_build_object(
+                    'conditions', t2.elem -> 'conditions',
+                    'routes', t2.elem -> 'routes',
+                    'name', t1.name
+                ) AS rule_with_name
+            FROM
+            (
+                (
+                    SELECT
+                        id,
+                        position,
+                        CONCAT('rule_', position::text) as name
+                    FROM
+                        router_versions,
+                        jsonb_array_elements(traffic_rules) with ordinality arr(elem, position)
+                ) AS t1
+                LEFT JOIN
```

Comment:
I don't think this LEFT JOIN is needed:

```sql
SELECT
    id,
    position,
    CONCAT('rule_', position::text) as name,
    elem
FROM
    router_versions2,
    jsonb_array_elements(traffic_rules) with ordinality arr(elem, position);
```

Reply:
In fact, I'd simplify the whole update query to this (still need to add in the null handling):
```sql
WITH tbl1 AS (
    SELECT id, position, CONCAT('rule_', position:: text) AS name, elem
    FROM router_versions2, jsonb_array_elements(traffic_rules) WITH ordinality arr(elem, position)
),
tbl2 AS (
    SELECT id, position, jsonb_build_object(
        'conditions', elem -> 'conditions',
        'routes', elem -> 'routes',
        'name', name
    ) AS traffic_rule FROM tbl1
),
tbl3 AS (
    SELECT id, json_agg(traffic_rule ORDER BY position) AS traffic_rules_updated
    FROM tbl2 GROUP BY id
)
UPDATE router_versions2 AS t SET traffic_rules=traffic_rules_updated
FROM tbl3
WHERE tbl3.id = t.id;
```

In general, I find the [CTE](https://www.geeksforgeeks.org/cte-in-sql/) query more readable.

Reply:
Updated the query